oxide-auth/src/primitives/registrar.rs

```diff
@@ -91,7 +92,7 @@ pub enum RegisteredUrl {
 /// possible if clients are allowed to provide any semantically matching URL as there are
 /// infinitely many with different hashes. (Note: a hashed form of URL storage is not currently
 /// supported).
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
 pub struct ExactUrl(String);
```

Comment:
There's an invariant for this field: it _must_ represent a valid URL. It's not memory-safety relevant, however, violating this can cause crashes later on as the URL is parsed with unwrap later. `serde` deserialization should check this property.